### PR TITLE
THNN functional module conversions batch 2

### PR DESCRIPTION
**Pull request by andreaskoepf**
- [x] ELU
- [x] LeakyReLU
- [x] LogSigmoid
- [x] LogSoftMax
- [x] LookupTable
- [x] Cross-check with old module implementation

### CODE REVIEW

File: lib/THNN/generic/THNN.h [hunk 1]
```diff
@@ -91,4 +104,50 @@ TH_API void THNN_(L1Cost_updateGradInput)(
...
+          THTensor *output);
+
+TH_API void THNN_(LookupTable_accGradParameters)(
+          THNNState *state,
+          THLongTensor *input,
+          THTensor *gradOutput,
+          THTensor *gradWeight,
+          real lr,
+          bool shouldScaleGradByFreq,
+          THLongTensor* count);
```
Comment (fmassa) [inline lib/THNN/generic/THNN.h]: Wouldn't it be better to use `THIndexTensor` instead ? You have changed everywhere from `int` to `long`, I thought you would want to make it consistent with `THIndexTensor`

File: lib/THNN/generic/LookupTable.c [hunk 2]
```diff
@@ -20,28 +20,23 @@ static void nn_(LookupTable_resetCount)(int *count_data, THLongTensor *input)
...
-
-  if (luaT_getfieldcheckboolean(L, 1, "shouldScaleGradByFreq"))
+  long *count_data = NULL;
+  
+  if (shouldScaleGradByFreq)
   {
-    THIntTensor *count = luaT_getfieldcheckudata(L, 1, "_count", "torch.IntTensor");
-    THIntTensor_resize1d(count, gradWeight->size[0]);
-    count_data = THIntTensor_data(count);
+    THLongTensor_resize1d(count, gradWeight->size[0]);
```
Comment (fmassa) [inline lib/THNN/generic/LookupTable.c]: Same thing here, wouldn't it be better to use `THIndexTensor_(resize1d)` ?

**Comment by fmassa** [comment]
There are a few places where you explicitely used `THLongTensor`, but I think it would be better to use `THIndexTensor`. Same thing for the `long` type, instead of `TH_index_t`

File: lib/THNN/generic/LookupTable.c [hunk 3]
```diff
@@ -2,7 +2,7 @@
 #define TH_GENERIC_FILE "generic/LookupTable.c"
 #else
 
-static void nn_(LookupTable_resetCount)(int *count_data, THLongTensor *input)
+void THNN_(LookupTable_resetCount)(long *count_data, THLongTensor *input)
```
Comment (andreaskoepf) [inline lib/THNN/generic/LookupTable.c]: Helper functions should be static.

File: LookupTable.lua [hunk 4]
```diff
@@ -82,7 +95,7 @@ function LookupTable:type(type, tensorCache)
       self._input = self.weight.new()
    else
       -- self._count and self._input should only be converted if using Cuda
-      self._count = torch.IntTensor()
+      self._count = torch.LongTensor()
```
Comment (szagoruyko) [inline LookupTable.lua]: I think it's a bad idea to change Int to Long here, might break something.

**Comment by andreaskoepf** [comment]
@szagoruyko OK, will restore IntTensor. I already discussed this with @fmassa. Ideally the THNN.h file would be usable for the CPU and GPU backend .. but that would mean we need some additional macro beside THIndexTensor that expands to IntTensor on CPU and CudaFloatTensor (or later CudaIntTensor) in GPU .. maybe something like THIntTensor?

**Comment by soumith** [comment]
Thanks Andreas!

**Comment by hughperkins** [comment]
Hmmm, this breaks `clnn` :-(

**Comment by hughperkins** [comment]
(breaks to clnn following this pull are fixed now, I think.  at least, tests pass now).

**Pull request merged**